### PR TITLE
Test question selection

### PR DESCRIPTION
**Pull request by asif-khan17**
**Fixed Issues**

#21359, #21929

**Files Changed**
test-question.component.ts: Modified
test-question.html: total number of questions field displayed
TestRepository.cs: Modified GetAllQuestionsByIdAsync(), GetTestByIdAsync() userId condition is added while fetching questions.
TestController.cs : Modified GetAllQuestionsByIdAsync(), GetTestByIdAsync() userId condition is added while fetching questions.
CategoryAc.cs: Modified QuestionCount field is added
TestRepositoryTest.cs:  Modified

**Checks**

- [X] Naming Conventions
- [X] Server side code comments (proper English, grammar and no spelling mistakes)
- [X] Client side code comments (proper English, grammar and no spelling mistakes)
- [X] Unused variables, methods, blank spaces and name spaces. 
- [X] Optimal Code and code formatting
- [X] Commit History is proper, no merge is done. 
- [X] Commit/pull request messages should be proper to justify your feature
- [X] All test cases are executed provided by tester.

**Known Issues**

**Comments**

NA

---
This change is [Reviewable](https://reviewable.io/reviews/promact/trappist/347)

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Repository/Test/ITestsRepository.cs [hunk 1]
```diff
@@ -104,7 +104,7 @@ public interface ITestsRepository
         /// <param name="testId">Parameter "testId" takes value of a Test's Id from route </param>
         /// <param name="CategoryId">Parameter "CategoryId" takes value of a Category's Id from route</param>
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Test/ITestsRepository.cs]: no xml comment for user id

File: Trappist/src/Promact.Trappist.Repository/Test/ITestsRepository.cs [hunk 2]
```diff
@@ -119,8 +119,8 @@ public interface ITestsRepository
         /// </summary>
         /// <param name="testId">Parameter "testId" takes value of a Test's Id from route</param>
         /// <returns>TestAC object</returns>       
-        Task<TestAC> GetTestByIdAsync(int testId);
-        
+        Task<TestAC> GetTestByIdAsync(int testId, string userId);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Test/ITestsRepository.cs]: same as above.

File: Trappist/src/Promact.Trappist.Repository/Test/ITestsRepository.cs [hunk 3]
```diff
@@ -104,7 +104,7 @@ public interface ITestsRepository
         /// <param name="testId">Parameter "testId" takes value of a Test's Id from route </param>
         /// <param name="CategoryId">Parameter "CategoryId" takes value of a Category's Id from route</param>
         /// <returns>List of QuestionAC objects of particular category</returns>      
-        Task<List<QuestionAC>> GetAllQuestionsByIdAsync(int testId, int categoryId);
+        Task<List<QuestionAC>> GetAllQuestionsByIdAsync(int testId, int categoryId, string userId);
```
Comment (iroshni) [inline Trappist/src/Promact.Trappist.Repository/Test/ITestsRepository.cs]: not updated in comment parameters.

File: Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs [hunk 4]
```diff
@@ -244,6 +244,7 @@ public TestsRepository(TrappistDbContext dbContext, IGlobalUtil util, IStringCon
                 var testCategoryList = await _dbContext.TestCategory.Where(x => x.TestId == testId).Include(x => x.Category).ToListAsync();
                 categoryListAc.ForEach(category =>
                 {
+                    category.QuestionCount = _dbContext.Question.Where(x => x.CategoryID == category.Id && x.CreatedByUserId == userId).ToList().Count();
```
Comment (iroshni) [inline Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs]: you can use count() method directly on where(). no need to do ToList(). adding ToList() increases overhead of firing query which count() will do by itself,

File: Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs [hunk 5]
```diff
@@ -238,16 +238,17 @@ public TestsRepository(TrappistDbContext dbContext, IGlobalUtil util, IStringCon
...
                 //Fetches the category list from Category Model
                 var categoryList = await _dbContext.Category.ToListAsync();
-                //Maps Category list to CategoryAC lis
+                //Maps Category list to CategoryAC list
                 var categoryListAc = Mapper.Map<List<Category>, List<CategoryAC>>(categoryList);
                 //Fetches the list of Categories from TestCategory Model
                 var testCategoryList = await _dbContext.TestCategory.Where(x => x.TestId == testId).Include(x => x.Category).ToListAsync();
                 categoryListAc.ForEach(category =>
                 {
+                    category.QuestionCount = _dbContext.Question.Where(x => x.CategoryID == category.Id && x.CreatedByUserId == userId).Count();
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs]: we can pass the predicate in count function itself.

File: Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs [hunk 6]
```diff
@@ -238,16 +238,17 @@ public TestsRepository(TrappistDbContext dbContext, IGlobalUtil util, IStringCon
...
                     //If category present in TestCategory Model,then its IsSelect property made true
                     if (testCategoryList.Exists(x => x.CategoryId == category.Id))
                     {
-                        category.NumberOfSelectedQuestion = _dbContext.TestQuestion.Where(x => x.Question.CategoryID == category.Id && x.TestId == testId).ToList().Count();
+                        category.NumberOfSelectedQuestion = _dbContext.TestQuestion.Where(x => x.Question.CategoryID == category.Id && x.TestId == testId).Count();
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs]: same as above.

**Pull request merged**